core/src/main/java/com/linecorp/armeria/server/healthcheck/FixedDelayHealthChecker.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the Licenses
+ */
+
+package com.linecorp.armeria.server.healthcheck;
+
+import java.time.Duration;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
+
+import com.linecorp.armeria.client.retry.Backoff;
+import com.linecorp.armeria.common.util.AbstractListenable;
+
+import io.netty.channel.EventLoop;
+
+/**
+ * Trigger supplied health checker after constructing and subsequently with the given delay between the
+ * termination of one execution and the commencement of the next.
+ */
+final class FixedDelayHealthChecker extends AbstractListenable<HealthChecker>
+        implements ListenableHealthChecker {
+    private final AtomicBoolean isHealthy = new AtomicBoolean(false);
+    private final Supplier<CompletionStage<Boolean>> healthChecker;
+    private final EventLoop eventLoop;
+    private final Backoff backoff;
+
+    FixedDelayHealthChecker(Supplier<CompletionStage<Boolean>> healthChecker, Duration delay, double jitter,
+                            EventLoop eventLoop) {
+        this.healthChecker = healthChecker;
+        this.eventLoop = eventLoop;
+        backoff = Backoff.fixed(delay.toMillis()).withJitter(jitter);
+
+        eventLoop.execute(createTask());
+    }
+
+    @Override
+    public boolean isHealthy() {
+        return isHealthy.get();
+    }
+
+    private void scheduleFetchTask() {
+        eventLoop.schedule(createTask(), backoff.nextDelayMillis(1), TimeUnit.MILLISECONDS);
```

Comment:
Don't we need to cancel the scheduled future when a server is stopping?
How about implement `SafeCloseable` and cancel the scheduled future in `close()` method and call `close() here?
https://github.com/line/armeria/blob/e4de28f17d2ffca291ca5a6d96e3801b5a4dac39/core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java#L304-L308

For example:
```java
if (c instanceof SafeCloseable) {
    (SafeCloseable c).close();
} 
```

Reply:
About this I'm curious if we can reply on evenloopgroup to do it. Like currently I use CommonPools.workerGroup() and it will be shutdowned by Server Config. 

But maybe it's better to provide custom close because sometimes it's not only canceling evenloop but other resources need to be cleared?

Reply:
>  Like currently I use CommonPools.workerGroup() and it will be shutdowned by Server Config.

As you know, the default value of `shutdownWorkerGroupOnStop` is false. So it would be better to close the scheduled future explicitly for avoiding unexpected leaks.

>  it's not only canceling evenloop but other resources need to be cleared?

Yes. It is possible. But I can't imagine yet what could be added. 😆